src/MQTTAsyncUtils.c

```diff
@@ -3073,7 +3109,7 @@ static MQTTPacket* MQTTAsync_cycle(int* sock, unsigned long timeout, int* rc)
 								data.token = command->command.token;
 								data.reasonCode = ackrc;
 								data.properties = msgprops;
-								data.packet_type = pack->header.bits.type;
+								data.packet_type = packet_type;
```

Comment:
Changing this to get the packet type from a local variable instead of from _pack->header.bits.type_ allows _pack_ to be freed at a point before this in a consistent manner. Other changes to use this variable above are for convenience/efficiency.